src/api/pricer/CoinGecko.ts

```diff
@@ -51,14 +52,16 @@ export default class CoinGecko extends BaseExchangeApi {
       throw Error(`CoinGecko API status ${response.status}`);
     }
 
-    if (!response[sourceCurrency]?.[currencyId]) {
+    if (!response[sourceCurrencyName]?.[targetCurrencyName]) {
       throw Error(
-        `Exchange rate for currency pair ${sourceCurrency}/${currencyId} is not available`
+        `Exchange rate for currency pair ${sourceCurrency}/${targetCurrency} is not available`
       );
     }
 
     // It cannot be undefined due to the previous check
-    const conversionRate = response[sourceCurrency]?.[currencyId] as string;
+    const conversionRate = response[sourceCurrencyName]?.[
+      targetCurrencyName
+    ] as string;
 
     return BigNumberJs(conversionRate);
```

Comment:
I'd like to point out that we're using bigNumberJs here even tho no calculations are yet being performed, so perhaps we should use BigNumber as return type.

Reply:
what are you referring? BigNumberJs or BigNumber from ethers, as return type? BigNumber from ethers does not handle decimals.

Reply:
ah of course, you are right.